src/Sylius/Bundle/PaymentBundle/Action/WebhookPaymentMethodAction.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PaymentBundle\Action;
+
+use Sylius\Bundle\PaymentBundle\Announcer\PaymentRequestAnnouncerInterface;
+use Sylius\Bundle\PaymentBundle\Provider\WebhookPaymentProviderInterface;
+use Sylius\Component\Payment\Factory\PaymentRequestFactoryInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentMethodRepositoryInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class WebhookPaymentMethodAction
+{
+    /**
+     * @param PaymentMethodRepositoryInterface<PaymentMethodInterface> $paymentMethodRepository
+     * @param PaymentRequestFactoryInterface<PaymentRequestInterface> $paymentRequestFactory
+     * @param PaymentRequestRepositoryInterface<PaymentRequestInterface> $paymentRequestRepository
+     */
+    public function __construct(
+        private PaymentMethodRepositoryInterface $paymentMethodRepository,
+        private PaymentRequestFactoryInterface $paymentRequestFactory,
+        private PaymentRequestRepositoryInterface $paymentRequestRepository,
+        private PaymentRequestAnnouncerInterface $paymentRequestAnnouncer,
+        private WebhookPaymentProviderInterface $webhookPaymentProvider,
+    ) {
+    }
+
+    public function __invoke(Request $request, string $code): Response
+    {
+        $paymentMethod = $this->paymentMethodRepository->findOneBy([
+            'code' => $code,
+        ]);
+
+        if (null === $paymentMethod) {
+            throw new NotFoundHttpException(sprintf('No payment method found with code "%s".', $code));
+        }
+
+        $payment = $this->webhookPaymentProvider->getPayment($request, $paymentMethod);
+
+        $paymentRequest = $this->paymentRequestFactory->create($payment, $paymentMethod);
+        $paymentRequest->setAction(PaymentRequestInterface::ACTION_NOTIFY);
+        $paymentRequest->setPayload($request);
+
+        $this->paymentRequestRepository->add($paymentRequest);
```

Comment:
I'm not up to date with payment request overall design, but as far as I see any http request to this controller/url will be automatically saved to DB?

How do I discard this request, if I don't consider it's correct, like signature verification at payum `NotifyNullAction`?

Reply:
L53:
```php
$payment = $this->webhookPaymentProvider->getPayment($request, $paymentMethod);
```

This previous line is asking to the developer what is the related payment contains in the request, then only if a payment is returned the request will be saved to the database during a message processing (syn or async).
Then the responsability rely on the payment method developer.

Do you see another way of handling this ?